### PR TITLE
Exposing Dynamic Analyzer REST APIs

### LINKED ISSUE #1552: [FEATURE] Expose Android Dynamic Analysis REST APIs
If you're requesting a new feature/enhancement, explain why you'd like it to be added and it's importance.

**Is your feature request related to a problem? Please describe.**
Expose Android Dynamic Analysis REST APIs

**Describe the solution you'd like**
Expose and document Dynamic Analysis APIs

### PR DESCRIPTION
**Pull request by ajinabraham**
### Describe the Pull Request

```
Exposing Dynamic Analyzer REST APIs

api/v1/dynamic/get_apps
api/v1/dynamic/start_analysis
api/v1/dynamic/stop_analysis
api/v1/dynamic/report_json
api/v1/dynamic/view_source
api/v1/android/logcat
api/v1/android/mobsfy
api/v1/android/adb_command
api/v1/android/root_ca
api/v1/android/activity
api/v1/frida/instrument
api/v1/frida/api_monitor
api/v1/frida/logs
api/v1/frida/list_scripts
api/v1/frida/get_script

Add REST API only mode with environment variable MOBSF_API_ONLY
```
Fixes #1552 

### Checklist for PR

- [x] Run MobSF unit tests and lint `tox -e lint,test`.
- [x] Tested Working on Linux, Mac, Windows, and Docker
- [x] Make sure build is passing on your PR. [Build Status](https://travis-ci.com/MobSF/Mobile-Security-Framework-MobSF/pull_requests)

### Additional Comments (if any)

```
DESCRIBE HERE
```

### CODE REVIEW

File: DynamicAnalyzer/views/android/dynamic_analyzer.py [hunk 1]
```diff
@@ -44,33 +55,41 @@ def dynamic_analysis(request):
...
-def dynamic_analyzer(request):
+def dynamic_analyzer(request, api=False):
     """Android Dynamic Analyzer Environment."""
     logger.info('Creating Dynamic Analysis Environment')
     try:
-        bin_hash = request.GET['hash']
-        package = request.GET['package']
+        if api:
+            bin_hash = request.POST['hash']
+            package = request.POST['package']
```
Comment (matandobr) [inline DynamicAnalyzer/views/android/dynamic_analyzer.py:83]: In various places the package is mandatory in addition to the hash, what is the reason for that?

**Review by matandobr — COMMENTED** [review]
Didn't finish going over the code, Tested the APIs on windows, works great!

File: DynamicAnalyzer/views/android/dynamic_analyzer.py (hunk 1 above)
Comment (ajinabraham) [reply DynamicAnalyzer/views/android/dynamic_analyzer.py:83]: I have just APIfied the existing APIs. We use the package name at some place like for example https://github.com/MobSF/Mobile-Security-Framework-MobSF/pull/1560/files#diff-c705086180b959833bcd533f2b3395b5ebda6e04bf40bae4dc07ec99ab9a4c1aR144 to verify the installation of apk, start main activity of the apk etc.

We use logcat filtering by package name 
https://github.com/MobSF/Mobile-Security-Framework-MobSF/pull/1560/files#diff-c705086180b959833bcd533f2b3395b5ebda6e04bf40bae4dc07ec99ab9a4c1aR209

md5 hash we have is only known to MobSF, it does not have any relevance inside the dynamic analyzer. It only knows about package name.

Comment (matandobr) [reply DynamicAnalyzer/views/android/dynamic_analyzer.py:83]: Yeah but the Dynamic analyzer could query for all the relevant info from the StaticAnalyzer.. isn't it better then making the user to make the query?

Comment (ajinabraham) [reply DynamicAnalyzer/views/android/dynamic_analyzer.py:83]: Make sense. I can update that.

**Comment by ajinabraham** [comment]
@matandobr Addressed the redundancy of package name where applicable.

**Review by matandobr — DISMISSED** [review]

File: DynamicAnalyzer/views/android/operations.py [hunk 2]
```diff
@@ -14,16 +14,33 @@
...
+from StaticAnalyzer.models import StaticAnalyzerAndroid
 
 logger = logging.getLogger(__name__)
 
 
 # Helpers
 
+def get_package_name(checksum):
+    """Get Package NAme from DB."""
+    static_android_db = StaticAnalyzerAndroid.objects.filter(
```
[the remark this answers is not in the document]
Comment (ajinabraham) [reply DynamicAnalyzer/views/android/operations.py:33]: Good point.

[the remark this answers is not in the document]
Comment (ajinabraham) [reply DynamicAnalyzer/views/android/operations.py:33]: I had fixed it here. In general, we use filter a lot. We need to replace that eventually.

**Pull request merged**